cosmo_tester/resources/scripts/create_replace_certs_config_script.py

```diff
@@ -27,19 +29,23 @@ def generate_replace_certs_config(replace_certs_config_path,
                     elif 'key' in cert_name:
                         node[cert_name] = key_path
             for key in replace_certs_config[instance_name]:
-                if 'ca' in key:
+                if 'ca' in key and 'key' not in key:
                     replace_certs_config[instance_name][key] = ca_path
+                if replace_ca_key and 'ca_key' in key:
+                    replace_certs_config[instance_name][key] = ca_key_path
     else:
         for instance_name, instance_dict in replace_certs_config.items():
             cert_path = certs_dir + host_ip + '.crt'
             key_path = certs_dir + host_ip + '.key'
             for cert_name in instance_dict:
-                if 'ca' in cert_name:
+                if 'ca' in cert_name and 'key' not in cert_name:
                     instance_dict[cert_name] = ca_path
                 elif 'cert' in cert_name:
                     instance_dict[cert_name] = cert_path
                 elif 'key' in cert_name:
                     instance_dict[cert_name] = key_path
+                elif replace_ca_key and 'ca_key' in cert_name:
```

Comment:
You'll never hit this branch, it'll always land on the previous one.

I suggest instead making the first branch handle all ca stuff- something like nesting below if 'ca' in cert_name: if 'ca_key' in cert_name: <something> else: <other thing>